components/google_drive/actions/add-file-sharing-preference/add-file-sharing-preference.js

```diff
@@ -0,0 +1,97 @@
+const googleDrive = require("../../google_drive.app");
+
+module.exports = {
+  key: "google_drive-add-file-sharing-preference",
+  name: "Add File Sharing Preference",
+  description:
+    "Add a sharing permission to the sharing preferences of a file and provide a sharing URL",
+  version: "0.0.1",
+  type: "action",
+  props: {
+    googleDrive,
+    /* eslint-disable pipedream/default-value-required-for-optional-props */
+    drive: {
+      propDefinition: [
+        googleDrive,
+        "watchedDrive",
+      ],
+      description:
+        "The drive to use. If not specified, your personal Google Drive will be used. If you are connected with any [Google Shared Drives](https://support.google.com/a/users/answer/9310351), you can select it here.",
```

Comment:
This (updated) description for the `drive` prop is repeated for many of the actions. The `drive` prop (including its base `description`) is used in other components. Should we move the `drive` prop to a `common` action file, is repeating the description OK, or should the descriptions be changed to be more action-specific?

Reply:
Updated descriptions for the drive prop in this commit: https://github.com/PipedreamHQ/pipedream/pull/1718/commits/bc5aa50090f97d06f61e1afb757f9b666ab7696c

- Made the `drive` prop optional, with default value `"My Drive"`
- Changed the base description for `drive` prop to a more general component-independent description
- Redefined description for the `drive` prop in Google Drive & Sheets sources
- Updated previously redefined `drive` prop description in Google Drive & Sheets actions

Alternatively, we could make **Drive** optional and additionally treat the value `undefined` as  "My Drive" so structured mode is enabled by default. For example:
<img width="387" alt="opt2-2" src="https://user-images.githubusercontent.com/19861096/137968330-40716da9-7f9b-4218-bd20-19253dccec9d.png">